packages/stdlib/src/async.js

```diff
@@ -0,0 +1,56 @@
+// @ts-check
+import { makePromise } from './makers';
+
+/**
+ * Resolve promise when await asyncPredThunk() returns falsy.
+ * 
+ * @template T
+ * @param {() => (T | Promise<T>)} asyncPredThunk Return truthy or a
+ * Promise<truthy> to keep looping.
+ * @returns {Promise<T>}
+ */
+const asyncWhile = asyncPredThunk => {
+  const executor = async (resolve, reject) => {
+    try {
+      const value = await asyncPredThunk();
```

Comment:
You preserved my violation of Jessie rules. We need to fix this. Do you see an easy fix?

Reply:
I implemented a fix without the try/catch, PTAL.